packages/graphback-cli/src/templates/configTemplates.ts

```diff
@@ -0,0 +1,49 @@
+import { writeFileSync } from 'fs';
+import { prompt as ask } from 'inquirer'
+import { Config } from '../templates/templateMetadata'
+
+/**
+ * Different database choices
+ */
+export const dbChoices: Config[] = [
+  {
+    name: 'postgres',
+    config: (projectName: string) => {
+      return {
+        'user': 'postgresql',
+        'password': 'postgres',
+        'database': `${projectName}`,
+        'host': '127.0.0.1',
+        'port': '5432'
+      }
+    }
+  }
+]
+
+
+/**
+ * Choose from the available db choices
+ */
+export async function chooseDB(projectName: string): Promise<object> {
+  const { dbName } = await ask([
+    {
```

Comment:
kinda pointless really. We can pick the name but we cannot really setup different port or something. How db will be running on localhost? Do we require people use docker compose before running CLI or CLI will deal with starting the database. How about error handling when database is not created?

Reply:
@wtrocki Should the user be able to customize about name, port, password etc ?

One way would be to custom generate a docker-compose file with user's inputs which can be later used by the db command before creating database.
Error handling would be handled in the db command.

Reply:
> Should the user be able to customize about name, port, password etc ? 

Yes. But we keep this file editable right in the project so we have this covered. 

Hmm.. we already have docker compose but this has some things hardcoded.
I think we can:

- have db abstraction later (mysql, postgress, mongo) and for the moment keep postgres, but to cather that IMHO let's move docker copose from template and inject it. So then we will have folder with databases and their starting config which is really nice way. 

Like we need this not only for database. The same requirement will go for templates. They are currently in the codebase of cmd but that is IMHO temporary. We need diagram of all moving pieces:

- Database of your choice
- Starting template of your choice
- Default client config
- Template user picks in that is missing all those and it is just runtime. 

This is just an idea and does not need to be executed now. 
The obvious drawback here is that we going to have this complexity of dropping different drivers into templates (currently it is all postgress) so another - much simpler option is to have also db dedicated templates, but this means a lots of replication of the code. 

So then we can have another idea:
Our templates execution layer can be abstracted to separate package and we can have templates that just share db specific code + typescript/Javascript etc.

All those probably should be spiked and we can play with this and see the best option.

Reply:
> > Should the user be able to customize about name, port, password etc ?
> 
> Yes. But we keep this file editable right in the project so we have this covered.
Wouldn't  the docker-compose files will be different for templates depending on their simplicity.